File: Leetcode/RemoveLinkedListElements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *removeElements(ListNode *head, int val)
    {
        if (head == NULL)
            return NULL;

        ListNode *tmp = head, *p;

        while (tmp->next != NULL)
        {
            if (tmp->next->val == val)
            {
                p = tmp->next;
                tmp->next = tmp->next->next;
                delete p;
            }
            else
                tmp = tmp->next;
        }
        while (head && head->val == val)
        {
            tmp = head;
            head = head->next;
            delete tmp;
        }
        return head;
    }
};

File: Leetcode/RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next


class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        tmp = ListNode(None)
        tmp.next = head
        prev = tmp
        crnt = head
        while crnt:
            if crnt.val == val:
                prev.next = crnt.next
                crnt = prev.next
            else:
                prev = prev.next
                crnt = crnt.next
        return tmp.next
